### PR TITLE
Emend spelling errors

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Subject: Fix typos found by codespell

### Feature or Bugfix
- Refactoring

### CODE REVIEW

File: sphinx/writers/text.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def cell_width(self, cell: Cell, source: list[int]) -> int:
         This takes into account cells spanning multiple columns.
         """
         if cell.row is None or cell.col is None:
-            msg = 'Cell co-ordinates have not been set'
+            msg = 'Cell coordinates have not been set'
```
Comment (AA-Turner) [inline sphinx/writers/text.py]: I prefer the hyphenated form, if possible

File: tests/test_extensions/test_ext_intersphinx_cache.py [hunk 2]
```diff
@@ -295,7 +295,7 @@ def test_load_mappings_cache_revert_update(tmp_path):
         app2.build()
         app2.cleanup()
 
-        # switch back to old url (re-use 'old_item')
+        # switch back to old url (reuse 'old_item')
```
Comment (AA-Turner) [inline tests/test_extensions/test_ext_intersphinx_cache.py:298]: Likewise

Comment (DimitriPapadopoulos) [reply tests/test_extensions/test_ext_intersphinx_cache.py:298]: Any dictionary contain the word reuse, so the usual grammar rule is that you don't add the hyphen as far as I know. Following grammar rules makes it easier for now-native speakers like me. But then all variants are OK, I cerrtainly can live with re-use.

Comment (AA-Turner) [reply tests/test_extensions/test_ext_intersphinx_cache.py:298]: Ok, I can be convinced on reuse -- let's leave as is in the PR

Comment (DimitriPapadopoulos) [reply tests/test_extensions/test_ext_intersphinx_cache.py:298]: I've already removed it. Besides, I see a preference for `re-use` among programmers nowadays, although `reuse` is more common in newspapers, books, etc. Languages evolve, there's always a gap between formal grammar rules and actual use, especially in English.

Comment (AA-Turner) [reply tests/test_extensions/test_ext_intersphinx_cache.py:298]: Ok, thank you. Please avoid force-pushing, we squash-merge all PRs.

**Review by AA-Turner — APPROVED** [review]

**Pull request merged**